test/new-e2e/tests/fips-compliance/fips_nix_test.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package fipscompliance
+
+import (
+	_ "embed"
+	"fmt"
+
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/provisioners/aws/host"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//go:embed fixtures/openssl-default.cnf
+var defaultOpenSSLConfig []byte
+
+type LinuxFIPSComplianceSuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+func TestLinuxFIPSComplianceSuite(t *testing.T) {
+	e2e.Run(t, &LinuxFIPSComplianceSuite{},
+		e2e.WithProvisioner(awshost.ProvisionerNoFakeIntake(
+			awshost.WithEC2InstanceOptions(ec2.WithOS(os.UbuntuDefault)),
+			awshost.WithAgentOptions(agentparams.WithFlavor("datadog-fips-agent")),
+		)))
+}
+
+func (v *LinuxFIPSComplianceSuite) TestFIPSDefaultConfig() {
+	status := v.Env().RemoteHost.MustExecute("sudo GOFIPS=0 datadog-agent status")
+	assert.NotContains(v.T(), status, "can't enable FIPS mode for OpenSSL")
+	assert.Contains(v.T(), status, "Status date")
+
+	status = v.Env().RemoteHost.MustExecute("sudo GOFIPS=1 datadog-agent status")
+	assert.NotContains(v.T(), status, "can't enable FIPS mode for OpenSSL")
+	assert.Contains(v.T(), status, "Status date")
+}
+
+func (v *LinuxFIPSComplianceSuite) TestFIPSNoFIPSProvider() {
+	v.Env().RemoteHost.MustExecute("sudo mv /opt/datadog-agent/embedded/ssl/openssl.cnf /opt/datadog-agent/embedded/ssl/openssl.cnf.tmp")
+	v.Env().RemoteHost.MustExecute(fmt.Sprintf(`sudo sh -c "echo '%s' > /opt/datadog-agent/embedded/ssl/openssl.cnf"`, defaultOpenSSLConfig))
```

Comment:
IIRC the goal of these line is to add `defaultOpenSSLConfig` in `/opt/datadog-agent/embedded/ssl/openssl.cnf` .
Have you tried to use `agentparams.WithFile` ? It should allow you to do what you want and using Pulumi make you sure that the file will be no longer present in the next test of your suite.
```suggestion
    v.UpdateEnv(e2e.WithProvisioner(awshost.ProvisionerNoFakeIntake(
			awshost.WithEC2InstanceOptions(ec2.WithOS(os.UbuntuDefault)),
			awshost.WithAgentOptions(
			     agentparams.WithFlavor("datadog-fips-agent"), 
                 agentparams.WithFile("/opt/datadog-agent/embedded/ssl/openssl.cnf", defaultOpenSSLConfig)),
		)))
```

Reply:
the `datadog-fips-agent` on install sets up the openssl fips-provider config at `/opt/datadog-agent/embedded/ssl/openssl.cnf` which is specific to the installation environment with self-tests

so what this doing is renaming that file and replacing it with the default configuration and then at the end of the test **moves it back** - iirc the WithFile function will rm the file when it is not in the next provisioner